packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -651,6 +653,14 @@ export interface CreateAlarmOptions {
    * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarm-evaluation
    */
   readonly datapointsToAlarm?: number;
+
+  /**
+   * In an alarm based on an anomaly detection model, this is the ID of the ANOMALY_DETECTION_BAND function used as the
+   * threshold for the alarm.
+   *
+   * @default - Not configured.
+   */
+  readonly thresholdMetricId?: string;
```

Comment:
This is somewhat under modeled, can this `Metric`? Which gets me to wonder if this is the right API. How will this look with `metric.CreateAlarm()` ?

Reply:
Not sure I follow on your first question, if you could elaborate please?

I've amended `metric.CreateAlarm()` [to pull the metric threshold ID](https://github.com/aws/aws-cdk/pull/10545/files#diff-0c15e1dbe524f9ba4c98b4faf9346abdR364) from `CreateAlarmOptions`. If I've missed any additional places this new property needs to be supported, let me know and I'll be happy to address that.